src/Microsoft.ML.Data/Transforms/Normalizer.cs

```diff
@@ -251,19 +251,23 @@ private static VersionInfo GetVersionInfo()
                 loaderAssemblyName: typeof(NormalizerTransformer).Assembly.FullName);
         }
 
-        private class ColumnInfo
+        public sealed class ColumnInfo
         {
             public readonly string Input;
             public readonly string Output;
             public readonly ColumnType InputType;
-            public readonly IColumnFunction ColumnFunction;
+            public readonly NormalizerModelParametersBase ModelParameters;
+            internal readonly IColumnFunction ColumnFunction;
 
-            public ColumnInfo(string input, string output, ColumnType inputType, IColumnFunction columnFunction)
+            private ColumnInfo() { }
```

Comment:
>ColumnInfo [](start = 20, length = 10)

why?

Reply:
Since this is public, so that users don;t instantiate it. 

---
In reply to: [234296093](https://github.com/dotnet/machinelearning/pull/1594#discussion_r234296093) [](ancestors = 234296093)

Reply:
I don't think it's needed: you already have a ctor, and a default parameterless public ctor is only generated if there's no ctor at all

---
In reply to: [234299425](https://github.com/dotnet/machinelearning/pull/1594#discussion_r234299425) [](ancestors = 234299425,234296093)